diff -Naur a/cmake/TPLs/FindTPLBinUtils.cmake b/cmake/TPLs/FindTPLBinUtils.cmake
--- a/cmake/TPLs/FindTPLBinUtils.cmake	2016-01-12 20:17:54.000000000 +0100
+++ b/cmake/TPLs/FindTPLBinUtils.cmake	2016-04-14 11:28:54.652116308 +0200
@@ -93,6 +93,7 @@
 #include <execinfo.h>
 #include <cxxabi.h>
 #include <link.h>
+#define PACKAGE Trilinos
 #include <bfd.h>
 
 
diff -Naur a/cmake/TPLs/FindTPLCCOLAMD.cmake b/cmake/TPLs/FindTPLCCOLAMD.cmake
--- a/cmake/TPLs/FindTPLCCOLAMD.cmake	2016-01-12 20:17:54.000000000 +0100
+++ b/cmake/TPLs/FindTPLCCOLAMD.cmake	2016-04-14 14:41:31.151107462 +0200
@@ -55,6 +55,6 @@
 
 
 TRIBITS_TPL_FIND_INCLUDE_DIRS_AND_LIBRARIES( CCOLAMD
-  REQUIRED_HEADERS ccolamd.h UFconfig.h
+  REQUIRED_HEADERS ccolamd.h
   REQUIRED_LIBS_NAMES ccolamd
   )
diff -Naur a/cmake/TPLs/FindTPLCholmod.cmake b/cmake/TPLs/FindTPLCholmod.cmake
--- a/cmake/TPLs/FindTPLCholmod.cmake	2016-01-12 20:17:54.000000000 +0100
+++ b/cmake/TPLs/FindTPLCholmod.cmake	2016-04-13 23:22:17.039116673 +0200
@@ -56,7 +56,7 @@
 
 TRIBITS_TPL_FIND_INCLUDE_DIRS_AND_LIBRARIES( Cholmod
   REQUIRED_HEADERS cholmod.h cholmod_core.h
-  REQUIRED_LIBS_NAMES libcholmod.a libamd.a libcolamd.a libsuitesparseconfig.a
+  REQUIRED_LIBS_NAMES cholmod amd colamd suitesparseconfig
   )
 
 
diff -Naur a/cmake/TPLs/FindTPLUMFPACK.cmake b/cmake/TPLs/FindTPLUMFPACK.cmake
--- a/cmake/TPLs/FindTPLUMFPACK.cmake	2016-01-12 20:17:54.000000000 +0100
+++ b/cmake/TPLs/FindTPLUMFPACK.cmake	2016-04-14 14:41:11.881107718 +0200
@@ -55,6 +55,6 @@
 
 
 TRIBITS_TPL_FIND_INCLUDE_DIRS_AND_LIBRARIES( UMFPACK
-  REQUIRED_HEADERS umfpack.h amd.h UFconfig.h
+  REQUIRED_HEADERS umfpack.h amd.h
   REQUIRED_LIBS_NAMES umfpack amd
   )
